crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -479,22 +479,82 @@ impl<'a> Resolver<'a> {
     }
 
     pub fn resolve_global_let(
-        &mut self, 
+        mut self, 
         let_stmt: crate::LetStatement
     ) -> (HirStatement, Vec<ResolverError>) {
-        match let_stmt.expression.kind {
-            ExpressionKind::Literal(_) => {
-                print!("*** 1\n");
-                let expression = self.resolve_expression(let_stmt.expression);
-                let definition = DefinitionKind::Global(expression);
-                (HirStatement::Let(HirLetStatement {
-                    pattern: self.resolve_pattern(let_stmt.pattern, definition),
-                    r#type: self.resolve_type(let_stmt.r#type),
-                    expression,
-                }), self.errors)
+        match let_stmt.expression.kind.clone() {
+            // Ensure it is a literal.
+            ExpressionKind::Literal(literal) => {
+                // If it is a literal array, check the elements are literals themselves.
+                match literal.clone() {
+                    // Check each element of a standard array.
+                    Literal::Array(ArrayLiteral::Standard(elems)) => {
+                        let mut type_error = false;
+                        for e in elems.iter() {
+                            match e.kind {
+                                ExpressionKind::Literal(_) => {}
```

Comment:
This can be ExpressionKind::Literal(_) => (),

Reply:
done!